lib/DirectoryWatcher.js

```diff
@@ -46,23 +46,13 @@ function DirectoryWatcher(directoryPath, options) {
 	this.path = directoryPath;
 	this.files = {};
 	this.directories = {};
-	this.watcher = chokidar.watch(directoryPath, {
-		ignoreInitial: true,
-		persistent: true,
-		followSymlinks: false,
-		depth: 0,
-		atomic: false,
-		alwaysStat: true,
-		ignorePermissionErrors: true,
+	this.watcher = sane(directoryPath, {
```

Comment:
According to [this repro repo](https://github.com/SpaceK33z/watchpack-pr-41-test) it seems like this can be called with non-directory or a non-existent directory. Sane will have to throw an error here because it's an obvious user error to call sane on a non-directory. See my comments:

- https://github.com/amasad/sane/issues/87
- https://github.com/amasad/sane/pull/88

Looking at the stack from the repro repo I think you should filter the files before mapping them to the `watcherManager` in [`watchpack.js`](https://github.com/webpack/watchpack/blob/08d4759606262c4216bf3b966163debb08553189/lib/watchpack.js#L35) which seems like it will solve the problem for you.